src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java

```diff
@@ -55,65 +58,104 @@
     }
 
     /**
-     * 
      * @param propertiesFile The location of the properties file.
-     * @param classLoader A classloader, useful if trying to programmatically configure with the daemon, such as in a
-     *        unit test.
-     * @param configurator A configurator to use.
-     * @throws IOException Thrown when the properties file can't be accessed.
+     * @param classLoader    A classloader, useful if trying to programmatically configure with the daemon, such as in a
+     *                       unit test.
+     * @param configurator   A configurator to use.
+     * @throws IOException              Thrown when the properties file can't be accessed.
      * @throws IllegalArgumentException Thrown when the contents of the properties file are not as expected.
      */
-    public MultiLangDaemonConfig(String propertiesFile,
-            ClassLoader classLoader,
+    public MultiLangDaemonConfig(String propertiesFile, ClassLoader classLoader,
             KinesisClientLibConfigurator configurator) throws IOException, IllegalArgumentException {
         Properties properties = loadProperties(classLoader, propertiesFile);
         if (!validateProperties(properties)) {
-            throw new IllegalArgumentException("Must provide an executable name in the properties file, "
-                    + "e.g. executableName = sampleapp.py");
+            throw new IllegalArgumentException(
+                    "Must provide an executable name in the properties file, " + "e.g. executableName = sampleapp.py");
         }
 
         String executableName = properties.getProperty(PROP_EXECUTABLE_NAME);
         String processingLanguage = properties.getProperty(PROP_PROCESSING_LANGUAGE);
+        ClientConfiguration clientConfig = buildClientConfig(properties);
+
+        kinesisClientLibConfig = configurator.getConfiguration(properties).withKinesisClientConfig(clientConfig)
+                .withCloudWatchClientConfig(clientConfig).withDynamoDBClientConfig(clientConfig);
 
-        kinesisClientLibConfig = configurator.getConfiguration(properties);
         executorService = buildExecutorService(properties);
-        recordProcessorFactory = new MultiLangRecordProcessorFactory(executableName, executorService, kinesisClientLibConfig);
+        recordProcessorFactory = new MultiLangRecordProcessorFactory(executableName, executorService,
+                kinesisClientLibConfig);
 
         LOG.info("Running " + kinesisClientLibConfig.getApplicationName() + " to process stream "
                 + kinesisClientLibConfig.getStreamName() + " with executable " + executableName);
         prepare(processingLanguage);
     }
 
+    private ClientConfiguration buildClientConfig(Properties properties) {
+        ClientConfiguration clientConfig = new ClientConfiguration();
+        String proxyHost = null;
+        int proxyPort = 0;
+
+        if (properties.getProperty("http.proxyHost") != null) {
```

Comment:
"http.proxyHost" and "http.proxyPort" can be extracted to a String, since it is being used in multiple places.

```java
String proxyHostProperty = "http.proxyHost";
String proxyPortProperty = "http.proxyPort";
String httpProxySystemProperty = "HTTP_PROXY";
```

Reply:
Yeah, I thought about that after.  I'll make them static an public so that I can reference them in the test as well.